src/include/migraphx/op/convolution.hpp

```diff
@@ -93,13 +91,6 @@ struct convolution
            x_shape.lens().at(1) != (w_shape.lens().at(1) * group))
             MIGRAPHX_THROW("CONVOLUTION: mismatched channel numbers");
 
-        std::vector<op::padding_mode_t> dyn_pad_modes = {op::padding_mode_t::same_upper,
-                                                         op::padding_mode_t::same_lower};
-        if(use_dynamic_same_auto_pad and not contains(dyn_pad_modes, padding_mode))
-        {
-            MIGRAPHX_THROW("CONVOLUTION: use_dynamic_same_auto_pad set with invalid padding mode");
-        }
```

Comment:
Do we need check that if inputs are dynamic then, padding must not be `default_` ?

Reply:
No, the padding can be constant when the inputs are dynamic. The `padding_mode` is only not `default_` when the user wants `SAME_LOWER` or `SAME_UPPER` padding with dynamic shapes.